### ISSUE TITLE
Raspberry Pi Fails on ofSystem (openFrameworks Core) error.

### ISSUE THREAD

**Issue by bakercp** [olab-io/ofSketch#58]
OfSketch  compiles on RPI and serves static content but WebSocket is closed prematurely. This may be due to an older version of poco, but more investigation is required.

**Comment by bakercp** [comment]
Will attempt to fix this with newer version of Poco.

**Comment by bakercp** [comment]
This seems to be related to ofxHTTP.  Funny thing is that the ofxHTTP and ofxJSONRPC examples run on RPI ...

**Comment by bakercp** [comment]
So, basically the situation is that everything compiles, but something is broken in the websocket communication.  Perhaps the WebSocket frames are too big or something -- I'm not sure yet.  I did just get this exception on the raspberry pi side, which is strange ....

```
[ error ] ServerWebSocketRouteHandler::handleRequest: exception: std::bad_alloc
[ error ] App::onWebSocketErrorEvent: Stop: 50
```

So, maybe that is a clue ...

**Comment by brannondorsey** [comment]
I don't have access to an RPi at the moment, but can we confirm that the browsers on the Pi fully support websockets?

**Comment by bakercp** [comment]
Yeah, it works under other circumstances.  I'm going to start injecting
logging info tonight to see if I can get it to reveal its mysteries ...

---

http://christopherbaker.net

**Comment by bakercp** [comment]
I have seen this error quite a few times in Chrome:

`Uncaught TypeError: Cannot read property 'isTemplate' of undefined`

I'm assuming that is a piece of javascript ... that perhaps is not being downloaded from the server or something?

**Comment by bakercp** [comment]
Alright I figured it out.

It all boils down to an error with `ofSystem` on Raspberry Pi.  This is called during the connection routine when `ofGetTargetPlatform()` is called.  This needs to be addressed in openFrameworks as an issue.  For the moment I am going to try to create a hack so we can keep moving.  You'll be happy to know as soon as I made my initial hack raspberry Pi worked splendidly (slow of course, but splendidly :)).
